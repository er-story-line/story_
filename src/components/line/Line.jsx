import React from 'react'
import PropTypes from 'prop-types'
import PostList from './PostList'

const Line = (props) => {
  const {
    title, index, postRepo, lineRepo,
  } = props

  return (
    <>
      <h1>{title}</h1>
      {index && (
        <PostList resource={index} postRepo={postRepo} lineRepo={lineRepo} />
      )}
    </>
  )
}

Line.propTypes = {
  title: PropTypes.string.isRequired,
  index: PropTypes.string.isRequired,
  lineRepo: PropTypes.shape({
    getMetadata: PropTypes.func.isRequired,
    getIndex: PropTypes.func.isRequired,
  }).isRequired,
  postRepo: PropTypes.shape({
    get: PropTypes.func.isRequired,
  }).isRequired,
}

export default Line
