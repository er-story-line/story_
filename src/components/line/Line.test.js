import React from 'react'
import renderer from 'react-test-renderer'
import Line from './Line'

jest.mock('./PostList', () => () => 'PostList')

describe('Line', () => {
  it('renders', () => {
    const tree = renderer.create(<Line title="My Line" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
