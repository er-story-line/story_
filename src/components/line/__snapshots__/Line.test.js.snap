// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line renders 1`] = `
Array [
  <h1>
    My Line
  </h1>,
  "PostList",
]
`;
